src/beacon-api/app/api/v1/operations.py

```
@@ -0,0 +1,214 @@
+    db: AsyncSession = Depends(get_async_db),
+    user_id: str = Depends(get_current_user_id)
+):
+    """Retrieve details of a specific device session."""
+    return await session_service.get_session(db, id)
+
+
+@router.get("/sessions", response_model=List[DeviceSessionResponse])
+async def list_sessions(
+    skip: int = Query(0, ge=0),
+    limit: int = Query(100, ge=1, le=1000),
+    db: AsyncSession = Depends(get_async_db),
```

Ruff (0.15.20)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)